File: Permutations/permutations.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {

    public static void generatePermutations(int[] nums, int ind, List<List<Integer>> ans) {
        if (ind == nums.length) {
            List<Integer> perm = new ArrayList<>();
            for (int num : nums) {
                perm.add(num);
            }
            ans.add(perm);
            return;
        }

        for (int i = ind; i < nums.length; i++) {
            int temp = nums[ind];
            nums[ind] = nums[i];
            nums[i] = temp;

            generatePermutations(nums, ind + 1, ans);

            temp = nums[ind];
            nums[ind] = nums[i];
            nums[i] = temp;
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter n: ");
        int n = sc.nextInt();

        int[] nums = new int[n];
        System.out.print("Enter nums: ");
        for (int i = 0; i < n; i++) {
            nums[i] = sc.nextInt();
        }

        List<List<Integer>> ans = new ArrayList<>();
        generatePermutations(nums, 0, ans);

        System.out.println("Permutations:");
        for (List<Integer> perm : ans) {
            for (int num : perm) {
                System.out.print(num + " ");
            }
            System.out.println();
        }

        sc.close();
    }
}

File: Permutations/permutations.cpp
#include <bits/stdc++.h>
using namespace std;

void gP(int ind, vector<int> &nums, vector<vector<int>> &ans)
{
    if (ind == nums.size())
    {
        ans.push_back(nums);
        return;
    }

    for (int i = ind; i < nums.size(); i++)
    {
        swap(nums[i], nums[ind]);
        gP(ind + 1, nums, ans);
        swap(nums[i], nums[ind]);
    }
}

int main()
{

    int n;
    cout << "Enter n: ";
    cin >> n;

    vector<int> nums(n, 0);
    vector<vector<int>> ans;

    cout << "Enter nums: ";

    for (int i = 0; i < n; i++)
    {
        cin >> nums[i];
    }

    gP(0, nums, ans);

    cout << "Permutations: " << endl;
    for (int i = 0; i < ans.size(); i++)
    {
        for (int j = 0; j < ans[i].size(); j++)
        {
            cout << ans[i][j] << " ";
        }
        cout << endl;
    }
}